src/main/java/app/sami/languageWeb/config/ApplicationConfig.java

```diff
@@ -0,0 +1,45 @@
+package app.sami.languageWeb.config;
+
+import app.sami.languageWeb.repos.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
+import org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
+@RequiredArgsConstructor
+public class ApplicationConfig {
```

Comment:
`JwtService` should not live in `config`. Probably in `auth`.

You have 2 choices when dealing with packages

### Layer approach
This is better for beginners as they learn the structure and makes it clear what goes where and which package depends on which, ie `controllers -> services -> (domain) -> repo`
```
controllers/
  dtos/
  ...
services/
  ...
repositories/
  ...
domain/
  ...
```

### Domain layer (preferred)
The idea here is to group files by domain, making it a lot easier to follow. If I want to have information about `Text`, I can simply go to the `texts` package and everything is there for me to reach wheras in the first approach you'd need to navigate multiple packages.
```
auth/
  services/
    RegisterNewUserService
    AuthenticateUserService
  dtos/
    ...
  AuthController
users/
  WhateverService
  UserController
  User
  UserRepository
texts/
config/
```